src/commands/LettuceMeetVisualizer.ts

```diff
@@ -0,0 +1,292 @@
+    event: {
+      id: string;
+      title: string;
+      description: string;
+      type: number;
+      pollStartTime: string;
+      pollEndTime: string;
+      maxScheduledDurationMins: number;
+      timeZone: string;
+      pollDates: string[];
+      isScheduled: boolean;
+      start: string;
+      end: string;
+      createdAt: string;
+      updatedAt: string;
+      user: {
+        id: string;
+      };
+      googleEvents: any[];
+      pollResponses: {
+        id: string;
+        user: {
+          __typename: string;
+          name: string;
+          email: string;
+          id: string;
+        };
+        availabilities: UserAvailability[];
+        event: {
+          id: string;
+        };
+      }[];
+    };
+  };
+}
+
+/**
+ * Availability of a user.
+ */
+interface UserAvailability {
+  start: string;
+  end: string;
+}
+
+interface AvailabilityData {
+  numAvailable: number;
+  peopleAvailable: string[];
+}
+
+
+/**
+ * Visualizes the Lettuce Meet board.
+ */
+export default class LettuceMeetVisualizer extends Command {
+  constructor(client: BotClient) {
+    const definition = new SlashCommandBuilder()
+      .setName('lettucemeetvisualizer')
+      .addStringOption((option) =>
+        option.setName('boardcode').setDescription('Lettuce Meet board code').setRequired(true))
+      .addStringOption((option) =>
+        option.setName('filterby').setDescription('Emails or names of the users to filter by').setRequired(false))
+      .addNumberOption((option) =>
+        option.setName('timeblock').setDescription('Time block in minutes').setRequired(false))
+      .setDescription('Visualizes lettuce meet board.');
+
+    super(client, {
+      name: 'lettucemeetvisualizer',
+      boardRequired: true,
+      enabled: true,
+      description: 'Visualizes lettuce meet board.',
+      category: 'Utility',
+      usage: client.settings.prefix.concat('lettucemeetvisualizer'),
+      requiredPermissions: ['SEND_MESSAGES'],
+    }, definition);
+  }
+
+  public async run(interaction: CommandInteraction): Promise<void> {
+    await super.defer(interaction);
+
+    // VpJg7 is the default board code.
+    const boardCode = interaction.options.getString('boardcode');
+    const filterBy = interaction.options.getString('filterby');
+    const timeBlock = interaction.options.getNumber('timeblock') || 60;
+
+    if (!boardCode) {
+      await interaction.editReply('Please provide a board code.');
+      return;
+    }
+
+    const regex = new RegExp(',+', 'g');
+    const parsedFilterBy = filterBy?.split(regex).map(key => key.trim());
+
+    const boardData = await this.getBoardData(boardCode);
+    let pollResponses = boardData.data.event.pollResponses;
+    if (parsedFilterBy) {
+      pollResponses = pollResponses.filter(response =>
+        parsedFilterBy.includes(response.user.name) || parsedFilterBy.includes(response.user.email));
+    }
+
+    let allPeople = pollResponses.map(response => response.user.name);
```

Comment:
Are we reassigning allPeople somewhere? If we aren't reassigning vars, we should be using const instead